package/MDAnalysis/coordinates/TRC.py

```diff
@@ -0,0 +1,399 @@
+        root, ext = os.path.splitext(self.filename)
+        self.trcfile = util.anyopen(self.filename)
+        if (ext[1:] == "trc") or (ext[1:] == "trj"):
+            self.compression = None
+        else:
+            self.compression = ext[1:]
+
+        # Read and calculate some information about the trajectory
+        self.traj_properties = self._read_traj_properties()
+
+        self._cache = {}
+        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+
+        self._reopen()
+        self.ts.dt = self.traj_properties["dt"]
+
+        self._read_frame(0)
+
+    @property
+    @cached('n_atoms')
+    def n_atoms(self):
+        """The number of atoms in one frame."""
+        try:
+            return self._read_atom_count()
+        except IOError:
+            return 0
+
+    def _read_atom_count(self):
+        n_atoms = self.traj_properties["n_atoms"]
+        return n_atoms
+
+    @property
+    @cached('n_frames')
+    def n_frames(self):
+        """The number of frames in the trajectory."""
+        try:
+            return self._read_frame_count()
+        except IOError:
+            return 0
+
+    def _read_frame_count(self):
+        n_frames = self.traj_properties["n_frames"]
+        return n_frames
+
+    def _frame_to_ts(self, frameDat, ts):
+        """Convert a frame to a :class:`TimeStep`"""
+        ts.frame = self._frame
+        ts.time = frameDat["time"]
+
+        ts.data['time'] = frameDat["time"]
+        ts.data['step'] = frameDat["step"]
+
+        ts.dimensions = frameDat["dimensions"]
+        ts.positions = frameDat["positions"]
+
+        #
+        # Convert the units
+        #
+        if self.convert_units:
+            if ts.has_positions:
+                self.convert_pos_from_native(ts.positions)
+            if ts.dimensions is not None:
+                self.convert_pos_from_native(ts.dimensions[:3])
+            if ts.has_velocities:
+                self.convert_velocities_from_native(ts.velocities)
+
+        return ts
+
+    def _read_traj_properties(self):
+        """
+        * Reads the number of atoms per frame (n_atoms)
+        * Reads the number of frames (n_frames)
+        * Reads the startposition of the timestep block
+          for each frame (l_blockstart_offset)
+        """
+
+        traj_properties = {}
+
+        #
+        # Check which of the supported blocks comes first win the trajectory
+        #
+        first_block = None
+        with util.anyopen(self.filename) as f:
+            for line in iter(f.readline, ''):
+                for blockname in self.SUPPORTED_BLOCKS:
+                    if (blockname == line.strip()) and (blockname != 'TITLE'):
+                        # Save the name of the first non-Title block
+                        # in the trajectory file
+                        first_block = blockname
+
+                if (first_block is not None):
+                    break  # First block found
+
+        #
```

Comment:
Single linebreak.

Reply:
I would prefer to keep a single empty line after the deeply indented block, even if a comment follows as I struggle to read source code that is too compressed. I will use the `black` tool to comply with the standards.